### PR TITLE
Add helpers for matching on resource properties and add 11 new cops using those helpers

### PR DESCRIPTION
**Pull request by tas50**
Add a helper for extracting properties and their values from a resource. It's a bit complex to do since the properties can be nested under various different conditionals. We have to recursively walk through the resource to find them.

Also adds a helper for extracting the string parameters from the property methods. That's also a bit funky because they can have variables in the text values which makes them a complex ast.

Once that's done add 11 new cops that use these helpers:

Chef/ResourceSetsInternalProperties
Chef/ResourceSetsNameProperty
Chef/ResourceWithNoneAction
Chef/ChocolateyPackageUninstallAction
Chef/LaunchdDeprecatedHashProperty
Chef/LocaleDeprecatedLcAllProperty
Chef/UserDeprecatedSupportsProperty
Chef/PowershellScriptExpandArchive
Chef/PowershellInstallPackage
Chef/PowershellInstallWindowsFeature
Chef/ShellOutToChocolatey

As an added bonus I've also backfilled some specs for cops that were added previously to get us closer to 100% coverage

### CODE REVIEW

File: lib/rubocop/cop/chef/correctness/resource_with_none_action.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module RuboCop
+  module Cop
+    module Chef
+      # The :nothing action is often typo'd as :nothing
```
Comment (tyler-ball) [inline lib/rubocop/cop/chef/correctness/resource_with_none_action.rb]:
```suggestion
      # The :nothing action is often typo'd as :none
```

**Comment by lamont-granquist** [comment]
you probably want to write some kind of tests for this one that are internal and specific to this helper, possibly using some kind of mock-cop.

**Comment by tas50** [comment]
Yep lots of specs are going to be necessary here. Incoming

File: lib/rubocop/cop/chef/deprecation/locale_lc_all_property.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module RuboCop
+  module Cop
+    module Chef
+      # The supports property was removed in Chef Infra Client 13 in favor of individual 'manage_home' and 'non_unique' properties.
```
Comment (lamont-granquist) [inline lib/rubocop/cop/chef/deprecation/locale_lc_all_property.rb]: copypasta error here i think

Comment (tas50) [reply lib/rubocop/cop/chef/deprecation/locale_lc_all_property.rb]: yep

**Pull request merged**